component/repo.go

```diff
@@ -1955,10 +1995,43 @@ func (c *RepoComponent) checkDeployPermission(ctx context.Context, repoType type
 	return &user, repo, deploy, nil
 }
 
-func (c *RepoComponent) DeployUpdate(ctx context.Context, repoType types.RepositoryType, namespace, name, currentUser string, deployID int64, req types.ModelRunReq) error {
-	_, _, deploy, err := c.checkDeployPermission(ctx, repoType, namespace, name, currentUser, deployID)
+func (c *RepoComponent) checkDeployPermissionForServerless(ctx context.Context, deployReq types.DeployActReq) (*database.User, *database.Repository, *database.Deploy, error) {
```

Comment:
only `admin` has permission to operate on serverless deployment

Reply:
add function to check admin